### PR TITLE
Enable setting limit for get_datapoints and get_datapoints_frame. Also update Jenkinsfile, assets module, and add some tests.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by cognite-cicd** [comment]
[Tests failed](https://jenkins.cognite.ai/job/github-builds/job/cognite-sdk-python/job/PR-84/2/testReport/).

#### .tests.v05.test_depthseries
```
../../.local/share/virtualenvs/ds_cognite-sdk-python_PR-84-LKKQAUXKXWO23J--gWWg9Z2/lib/python3.6/site-packages/_pytest/python.py:469: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
../../.local/share/virtualenvs/ds_cognite-sdk-python_PR-84-LKKQAUXKXWO23J--gWWg9Z2/lib/python3.6/site-packages/py/_path/local.py:668: in pyimport
    __import__(modname)
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:656: in _load_unlocked
    ???
<frozen importlib._bootstrap>:626: in _load_backward_compatible
    ???
../../.local/share/virtualenvs/ds_cognite-sdk-python_PR-84-LKKQAUXKXWO23J--gWWg9Z2/lib/python3.6/site-packages/_pytest/assertion/rewrite.py:226: in load_module
    py.builtin.exec_(co, mod.__dict__)
tests/v05/test_depthseries.py:7: in <module>
    from cognite.v05 import depthseries, dto
cognite/v05/depthseries.py:19: in <module>
    import cognite.v05.timeseries as ts
E     File "/home/jenkins/workspace/ds_cognite-sdk-python_PR-84-LKKQAUXKXWO23JCRDGMJNNK4RO3CVPUV7ZQ6FZDW4HQNFEKWCW5Q/cognite/v05/timeseries.py", line 863
E       '''Generator function which continously polls latest datapoint of a timeseries and yields new datapoints.
E   
E       Args:
E           name (str): Name of timeseries to get latest datapoints for.
E   	
E   	update_frequency (float): Frequency to pull for data in seconds.
E   
E       Keyword Args:
E   
E           api_key (str): Your api-key.
E   
E           project (str): Project name.
E   
E       Yields:
E           dict: Dictionary containing timestamp and value of latest datapoint.
E       """
E       api_key, project = config.get_config_variables(kwargs.get("api_key"), kwargs.get("project"))
E       last_timestamp = get_latest(name, api_key=api_key, project=project).to_json()["timestamp"]
E       while True:
E           latest = get_latest(name, api_key=api_key, project=project).to_json()
E           if last_timestamp == latest['timestamp']:
E               time.sleep(update_frequency)
E           else:
E               yield latest
E           last_timestamp = latest["timestamp"]
E                                                                                                                
E   
E            
E                                                                       
E   	
E   	                                                                
E   
E                    
E   
E                                       
E   
E                                       
E   
E              
E                                                                               
E          
E                                                                                                   
E                                                                                                 
E                  
E                                                                                
E                                                    
E                                           
E                
E                           
E                                              ^
E   SyntaxError: EOF while scanning triple-quoted string literal
```

#### .tests.v05.test_timeseries
```
../../.local/share/virtualenvs/ds_cognite-sdk-python_PR-84-LKKQAUXKXWO23J--gWWg9Z2/lib/python3.6/site-packages/_pytest/python.py:469: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
../../.local/share/virtualenvs/ds_cognite-sdk-python_PR-84-LKKQAUXKXWO23J--gWWg9Z2/lib/python3.6/site-packages/py/_path/local.py:668: in pyimport
    __import__(modname)
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:656: in _load_unlocked
    ???
<frozen importlib._bootstrap>:626: in _load_backward_compatible
    ???
../../.local/share/virtualenvs/ds_cognite-sdk-python_PR-84-LKKQAUXKXWO23J--gWWg9Z2/lib/python3.6/site-packages/_pytest/assertion/rewrite.py:226: in load_module
    py.builtin.exec_(co, mod.__dict__)
tests/v05/test_timeseries.py:7: in <module>
    from cognite.v05 import dto, timeseries
E     File "/home/jenkins/workspace/ds_cognite-sdk-python_PR-84-LKKQAUXKXWO23JCRDGMJNNK4RO3CVPUV7ZQ6FZDW4HQNFEKWCW5Q/cognite/v05/timeseries.py", line 863
E       '''Generator function which continously polls latest datapoint of a timeseries and yields new datapoints.
E   
E       Args:
E           name (str): Name of timeseries to get latest datapoints for.
E   	
E   	update_frequency (float): Frequency to pull for data in seconds.
E   
E       Keyword Args:
E   
E           api_key (str): Your api-key.
E   
E           project (str): Project name.
E   
E       Yields:
E           dict: Dictionary containing timestamp and value of latest datapoint.
E       """
E       api_key, project = config.get_config_variables(kwargs.get("api_key"), kwargs.get("project"))
E       last_timestamp = get_latest(name, api_key=api_key, project=project).to_json()["timestamp"]
E       while True:
E           latest = get_latest(name, api_key=api_key, project=project).to_json()
E           if last_timestamp == latest['timestamp']:
E               time.sleep(update_frequency)
E           else:
E               yield latest
E           last_timestamp = latest["timestamp"]
E                                                                                                                
E   
E            
E                                                                       
E   	
E   	                                                                
E   
E                    
E   
E                                       
E   
E                                       
E   
E              
E                                                                               
E          
E                                                                                                   
E                                                                                                 
E                  
E                                                                                
E                                                    
E                                           
E                
E                           
E                                              ^
E   SyntaxError: EOF while scanning triple-quoted string literal
```

**Comment by cognite-cicd** [comment]
[Tests failed](https://jenkins.cognite.ai/job/github-builds/job/cognite-sdk-python/job/PR-84/3/testReport/).

#### TestDepthseries.test_post_depthseries
```
self = <tests.v05.test_depthseries.TestDepthseries object at 0x7f836b490828>

    def test_post_depthseries(self):
        tso = dto.TimeSeries(DS_NAME)
>       res = depthseries.post_depth_series([tso])

tests/v05/test_depthseries.py:24: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cognite/v05/depthseries.py:259: in post_depth_series
    raise e
cognite/v05/depthseries.py:244: in post_depth_series
    _utils.post_request(url, body=body, headers=headers)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

url = 'https://api.cognitedata.com/api/0.5/projects/mltest/timeseries'
body = {'items': []}
headers = {'accept': 'application/json', 'api-key': 'NDkxODdjZDYtNjcyMS00YjE1LTk4NzMtNGYyYmJkOGM3Y2Rj', 'content-type': 'application/json'}
params = None, cookies = None, use_gzip = False

    def post_request(url, body, headers=None, params=None, cookies=None, use_gzip=False):
        """Perform a POST request with a predetermined number of retries."""
        _log_request("POST", url, body=body, params=params, headers=headers, cookies=cookies)
    
        for number_of_tries in range(config.get_number_of_retries() + 1):
            try:
                if use_gzip:
                    if headers:
                        headers["Content-Encoding"] = "gzip"
                    else:
                        headers = {"Content-Encoding": "gzip"}
                    res = requests.post(
                        url,
                        data=gzip.compress(json.dumps(body).encode("utf-8")),
                        headers=headers,
                        params=params,
                        cookies=cookies,
                    )
                else:
                    res = requests.post(url, data=json.dumps(body), headers=headers, params=params, cookies=cookies)
                if res.status_code == 200:
                    return res
            except Exception as e:
                if number_of_tries == config.get_number_of_retries():
                    raise APIError(e)
        try:
            err_mess = res.json()["error"].__str__()
        except:
            err_mess = res.content.__str__()
        err_mess += "\nX-Request_id: {}".format(res.headers.get("X-Request-Id"))
>       raise APIError(err_mess)
E       cognite._utils.APIError: {'code': 400, 'message': 'Error: missing object: items array'}
E       X-Request_id: 71f35f9dcdca1bb8f8cf7a1b30fc82ba, 71f35f9dcdca1bb8f8cf7a1b30fc82ba

cognite/_utils.py:91: APIError
```

**Comment by erlendcognite** [comment]
Most of the changes here is reformatting. But that is in a separate commit. Also made some small changes to assets module.

File: cognite/v05/timeseries.py [hunk 1]
```diff
@@ -146,68 +153,109 @@ def _get_datapoints_helper(name, aggregates=None, granularity=None, start=None,
...
+    """
+    api_key, project = config.get_config_variables(kwargs.get("api_key"), kwargs.get("project"))
+    url = config.get_base_url(api_version=0.5) + "/projects/{}/timeseries/data/{}".format(project, quote(name, safe=""))
+
+    use_protobuf = kwargs.get("protobuf", True) and aggregates is None
+
+    params = {"aggregates": aggregates, "granularity": granularity, "limit": limit, "start": start, "end": end}
+
+    headers = {"api-key": api_key, "accept": "application/protobuf" if use_protobuf else "application/json"}
+    res = _utils.get_request(url, params=params, headers=headers)
```
Comment (tuanng-cognite) [inline cognite/v05/timeseries.py]: So, we use our API functionality to limit the number of datapoints? If this is the case, max limit is 100k. Then we should explicitly say that in the docstring of this method or caller method.

Comment (henrikCognite) [reply cognite/v05/timeseries.py]: And is this handled in this method when it’s not called with a limit? What happens if one of the intervals contains more that this threshold (100k)?

Comment (erlendcognite) [reply cognite/v05/timeseries.py]: @tuanng-cognite ok, will update the docstrings.
@henrikCognite this method is only used if the user passes limit to get_datapoints(). If not, it splits the request into equal time intervals and parallellizes the execution of _get_datapoints_helper() method which pages automatically within the each interval.

**Pull request merged**